wagtail/admin/active_filters.py

```diff
@@ -0,0 +1,208 @@
+    def get_active_filters(self):
+        field: ModelChoiceField = self.filter.field
+        yield from (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                field.label_from_instance(item),
+                self.get_url_without_filter_param_value(self.bound_field.name, item.pk),
+            )
+            for item in self.value
+        )
+
+
+class RangeFilterAdapter(BaseFilterAdapter):
+    def format_value(self, value):
+        return str(value) if value is not None else ""
+
+    def get_active_filters(self):
+        start_date_display = self.format_value(self.value.start)
+        end_date_display = self.format_value(self.value.stop)
+        widget = self.filter.field.widget
+        yield (
+            ActiveFilter(
+                self.bound_field.auto_id,
+                self.filter.label,
+                f"{start_date_display} - {end_date_display}",
+                self.get_url_without_filter_param(
+                    [
+                        widget.suffixed(self.bound_field.name, suffix)
+                        for suffix in widget.suffixes
+                    ]
+                ),
+            )
+        )
+
+
+class DateFromToRangeFilterAdapter(RangeFilterAdapter):
+    def format_value(self, value):
+        return date_format(value) if value is not None else ""
+
+
+filter_adapter_class_registry = ObjectTypeRegistry()
+
+
+def register_filter_adapter_class(
+    filter_class: django_filters.Filter,
+    adapter_class: BaseFilterAdapter = None,
+    exact_class: BaseFilterAdapter = False,
+):
+    """
+    Define how a django-filter Filter should be displayed when it's active.
+    The ``adapter_class`` should be a subclass of
+    ``wagtail.admin.active_filters.BaseFilterAdapter``.
+
+    This is mainly useful for defining how active filters are rendered in
+    listing views.
+    """
+    if adapter_class is None:
+        raise ImproperlyConfigured(
+            "register_filter_adapter_class must be passed a 'adapter_class' keyword argument"
+        )
+
+    filter_adapter_class_registry.register(
+        filter_class,
+        value=adapter_class,
+        exact_class=exact_class,
+    )
+
+
+register_filter_adapter_class(
+    django_filters.Filter,
+    BaseFilterAdapter,
+)
+register_filter_adapter_class(
+    django_filters.ChoiceFilter,
+    ChoiceFilterAdapter,
+)
+register_filter_adapter_class(
+    django_filters.MultipleChoiceFilter,
+    MultipleChoiceFilterAdapter,
+)
+register_filter_adapter_class(
+    django_filters.ModelChoiceFilter,
+    ModelChoiceFilterAdapter,
+)
+register_filter_adapter_class(
+    django_filters.ModelMultipleChoiceFilter,
+    ModelMultipleChoiceFilterAdapter,
+)
+register_filter_adapter_class(
+    django_filters.RangeFilter,
+    RangeFilterAdapter,
+)
+register_filter_adapter_class(
+    django_filters.DateFromToRangeFilter,
+    DateFromToRangeFilterAdapter,
+)
```

Comment:
DX: Do we want to handle other possible `Filter` classes by default? 

Missing ones are:
- `BooleanFilter` (would result in untranslated "yes", "no" or "unknown")
- `TypedChoiceFilter` (which can reuse the `ChoiceFilterAdapter`)
- `DateFilter`, `DateTimeFilter`, `IsoDateTimeFilter` and `TimeFilter` (would result in a date not formatted with `date_format()` or `time_format()`
- `DurationFilter` (could use `timesince()` or `timeuntil()`
- `NumericRangeFilter` (which can reuse the `RangeFilterAdapter`)
- `BaseCSVFilter` (I have not used that one, so unsure)
- `LookupChoiceFilter` (which could be dynamic depending on `field_class`)

Reply:
Ideally yes, but I think that could be tackled in a separate issue. For now, this PR is worth for the refactoring alone as it allows devs to add custom adapters in their code that they can then contribute back to Wagtail.